cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1407,6 +1407,10 @@ FORCE_INLINE
 const char* convert_fromUTF8_binary(gdv_int64 context, const char* bin_in, gdv_int32 len,
                                     gdv_int32* out_len) {
   *out_len = len;
+  // handle empty string case
```

Comment:
Here malloc and memcpy are not required. You can just return the original string slice. Also malloc returning nullptr for zero length in certain cases is also a bug. Can you create a separate PR to fix that.

Reply:
Made the changes, I'll be creating another PR for the malloc issue as well